smart-contract-verifier/smart-contract-verifier-proto/proto/v2/smart-contract-verifier.proto

```diff
@@ -250,4 +254,118 @@ message LookupMethodsResponse {
     uint32 length = 3;
   }
   map<string, Method> methods = 1;
-}
\ No newline at end of file
+}
+
+/*************** Batch related messages ***************/
+
+message BatchVerifySolidityMultiPartRequest {
+  /// Contracts to be verified
+  repeated Contract contracts = 1;
+
+  /// Compiler version used to compile the contract
+  string compiler_version = 2;
+  /// Map from a source file name to the actual source code
+  map<string, string> sources = 3;
+  /// Version of the EVM to compile for. If absent results in default EVM version
+  optional string evm_version = 4;
+  /// If present, optimizations are enabled with specified number of runs,
+  /// otherwise optimizations are disabled
+  optional uint32 optimization_runs = 5;
+  /// Map from a library name to its address
+  map<string, string> libraries = 6;
+}
+
+message BatchVerifySolidityStandardJsonRequest {
+  /// Contracts to be verified
+  repeated Contract contracts = 1;
+
+  /// Compiler version used to compile the contract
+  string compiler_version = 2;
+  /// https://docs.soliditylang.org/en/latest/using-the-compiler.html#input-description
+  string input = 3;
+}
+
+message BatchVerifyResponse {
+  message ContractVerificationResults {
+    repeated ContractVerificationResult items = 1;
+  }
+
+  oneof verification_result {
+    CompilationFailure compilation_failure = 1;
+    ContractVerificationResults contract_verification_results = 2;
+  }
+}
+
+message Contract {
+  optional string creation_code = 1;
+  optional string runtime_code = 2;
+  /// An optional field with metadata used for logging purposes
+  optional VerificationMetadata metadata = 3;
+}
+
+message CompilationFailure {
+  string message = 1;
+}
+
+message ContractVerificationResult {
+  oneof verification_result {
+    ContractVerificationFailure failure = 1;
+    ContractVerificationSuccess success = 2;
+  }
+}
+
+message ContractVerificationSuccess {
+  string creation_code = 1;
+  string runtime_code = 2;
+
+  message Compiler {
+    enum Compiler {
+      UNSPECIFIED = 0;
+      SOLC = 1;
+      VYPER = 2;
+    }
+  }
+
+  Compiler.Compiler compiler = 3;
+
+  string compiler_version = 4;
+
+  message Language {
+    enum Language {
+      UNSPECIFIED = 0;
+      SOLIDITY = 1;
+      YUL = 2;
+      VYPER = 3;
+    }
+  }
```

Comment:
why `Language.Language` ?

Reply:
maybe remove `message Language` ?

Reply:
Without message wrapper there is a conflict between `Language.VYPER` and `Compiler.VYPER` values. Thus, another option is to add `LANGUAGE_` prefix to all values which I personally like less than this approach